### PR TITLE
[DET-2271] feat: how-to use trained model checkpoints guide

### PR DESCRIPTION
**Pull request by sidneyw**
# Test Plan
- UI change:
  - [ ] add screenshots
  - [ ] React? build & check storybooks
- [ ] user-facing api change: modify documentation and examples
- [ ] user-facing api change: add the "User-facing API Change" label
- [ ] bug fix: add regression test
- [ ] bug fix: determine if there are other similar bugs in the codebase
- [ ] new feature: add test coverage for any user-facing aspects
- [ ] refactor: maintain existing code coverage

### CODE REVIEW

File: docs/how-to/use-trained-models.txt [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+.. code:: yaml
+
+   searcher:
+     metric: "loss"
+     smaller_is_better: true
+
+The following snippet will return an instance of :class:`Checkpoint
+<determined.experimental.Checkpoint>` representing the checkpoint from an
+experiment with the best validation metric.
```
Comment (katport) [inline docs/how-to/use-trained-models.txt:28]: let's explain here that this should be in any python script after a checkpoint has been made.

File: docs/how-to/use-trained-models.txt [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+Querying Checkpoints
+--------------------
+
+The :class:`ExperimentReference <determined.experimental.ExperimentReference>`
+class contains the ``top_checkpoint`` method. Without arguments the method will
+check the experiment configuration searcher field for the ``metric`` and
+``smaller_is_better`` values. These values are used to sort the experiments
+checkpoints by validation performance. For example the following searcher
+configuration will result in `ExperimentReference.top_checkpoint` returning the
+checkpoint with the lowest "loss" metric.
```
Comment (katport) [inline docs/how-to/use-trained-models.txt:18]: I think we need to add a sentence here to clarify that this will be in their experiment configuration file. 

Also, "without arguments" and "for example" are missing a comma.

File: docs/how-to/use-trained-models.txt [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+.. code:: python
+
+  from determined.experimental import Experiment
+
+  checkpoint = Experiment(id).top_checkpoint()
+
+To sort checkpoints by another metric, pass ``smaller_is_better`` and
+``sort_by`` parameters to ``top_checkpoint`` where ``sort_by`` is the name of
+the metric to sort by and ``smaller_is_better`` defines whether to minimize or
+maximize the metric.
```
Comment (katport) [inline docs/how-to/use-trained-models.txt:39]: Well, I can sort by any metric besides smaller_is_better right?

maybe clarify that by default it uses smaller_is_better ect.

Something like:
"Checkpoints can be sorted by any metric, by using "sort_by", which states which metric to use, and "smaller_is_better", which defines whether to minimize or maximize the metric."

File: docs/how-to/use-trained-models.txt [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+.. code:: python
+
+  from determined.experimental import Experiment
+
+  checkpoint = Experiment(id).top_checkpoint()
+  checkpoint_path = checkpoint.download()
+
+The ``load`` method of the :class:`Checkpoint
+<determined.experimental.Checkpoint>` class downloads the checkpoint if it
+does not already exist on locally and then load it into memory in a python
```
Comment (katport) [inline docs/how-to/use-trained-models.txt:110]: "does not exist locally"

File: docs/how-to/use-trained-models.txt [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+PyTorch checkpoints are saved via pickle and loaded as ``nn.Module``
+(see `pytorch serialization <https://pytorch.org/docs/stable/notes/serialization.html>`_ for details).
+
+Download Checkpoints via the CLI
+--------------------------------
+Determined offers a CLI command, ``det trial download`` for downloading checkpoints locally.
+Similar to the :class:`TrialReference <determined.experimental.TrialReference>`
+above, the command accepts ``--best``, ``--latest``, and ``--uuid`` options.
+
+.. code:: bash
```
Comment (katport) [inline docs/how-to/use-trained-models.txt:132]: remove bash

**Comment by sidneyw** [comment]
See #289
